lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -740,6 +738,27 @@ func (c *Client) sharedDirectoryWriteRequest(req tdp.SharedDirectoryWriteRequest
 	return C.ErrCodeFailure
 }
 
+//export tdp_sd_move_request
+func tdp_sd_move_request(handle C.uintptr_t, req *C.CGOSharedDirectoryMoveRequest) C.CGOErrCode {
+	return cgo.Handle(handle).Value().(*Client).sharedDirectoryMoveRequest(tdp.SharedDirectoryMoveRequest{
+		CompletionID: uint32(req.completion_id),
+		DirectoryID:  uint32(req.directory_id),
+		OriginalPath: C.GoString(req.original_path),
+		NewPath:      C.GoString(req.new_path),
+	})
+}
+
+func (c *Client) sharedDirectoryMoveRequest(req tdp.SharedDirectoryMoveRequest) C.CGOErrCode {
+	if c.cfg.AllowDirectorySharing {
+		if err := c.cfg.Conn.OutputMessage(req); err != nil {
+			c.cfg.Log.Errorf("failed to send SharedDirectoryMoveRequest: %v", err)
+			return C.ErrCodeFailure
+		}
+		return C.ErrCodeSuccess
+	}
+	return C.ErrCodeFailure
```

Comment:
nit: For me this looks cleaner:
```suggestion
	if !c.cfg.AllowDirectorySharing {
		return C.ErrCodeFailure		
	}

	if err := c.cfg.Conn.OutputMessage(req); err != nil {
		c.cfg.Log.Errorf("failed to send SharedDirectoryMoveRequest: %v", err)
		return C.ErrCodeFailure
	}
	return C.ErrCodeSuccess
```

Reply:
I have a bunch of these so I fixed them all 45e8058b3d481e0b25fe7b37ac099e7cb1347a3a